### PR TITLE
Restart handler modernization

### PR DESCRIPTION
**Pull request by mtju**
Restart handler modernization and a note about automatic expansion. Please take a look at commit messages for more information.

### CODE REVIEW

File: handlers/main.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
 - name: Restart
-  command: /sbin/shutdown -r +0 'Ansible Reboot triggered'
-  async: 0
+  shell: sleep 2 && /sbin/shutdown -r now 'Ansible Reboot triggered'
```
Comment (charleskorn) [inline handlers/main.yml:3]: What's the reason for waiting two seconds before restarting?

Comment (mtju) [reply handlers/main.yml:3]: Refer to this comment https://github.com/ansible/ansible/issues/14413#issuecomment-182470142

Comment (charleskorn) [reply handlers/main.yml:3]: OK, strange but makes sense...

**Comment by charleskorn** [comment]
Thanks for the PR!

**Pull request merged**